typing_tests/pine-typed-result.ts

```diff
@@ -473,23 +506,25 @@ export let aStringOrUndefined: string | undefined;
 }
 
 {
-	type deviceOptionsExpandNavigationResourceString = PineClient.TypedResult<
-		AnyObject,
-		{
-			$select: 'id';
-			$expand: {
-				is_pinned_on__release: {
-					$select: 'commit';
+	type deviceOptionsExpandNavigationResourceString = NonNullable<
+		BalenaSdk.Pine.OptionsToResponse<
+			AnyObject,
+			{
+				$select: 'id';
+				$expand: {
+					is_pinned_on__release: {
+						$select: 'commit';
+					};
 				};
-			};
-		}
+			},
+			number
+		>
 	>;
 
 	const result: deviceOptionsExpandNavigationResourceString = {} as any;
 
 	aNumber = result.id;
 	// Errors, since it could be an OptionalNavigationResource
-	// @ts-expect-error test case
 	aStringOrUndefined = result.is_pinned_on__release[0].commit;
```

Comment:
Same as earlier, add a todo that this is nullable and should complain since we are not using `?.[0]`